cmd/az-analyze/cmd/azva_test.go

```diff
@@ -0,0 +1,317 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1beta1 "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1beta1"
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+)
+
+var azvaResourceAll0 = AzvaResource{
```

Comment:
Now that we have removed AzvaResourceAll and are using AzvaResource for all the scenarios, do we really need to have different azvaResource by pod, zone, node, all etc? Can't we have a few AzvaResource objects that have the information for pods,nodes,zones? We can certainly verify the result based on what we are testing. Like if we are testing GetAzvaByZone, we would only match the zone. Same for pod and node.

Reply:
Looking more closely at it, the current implementation seems fine to me. Because the actual output will have the fields empty. I'll resolve this.

Reply:
If the result is from GetAzvaByZone, its PodName and NodeName should be empty string. So don't we check that?

Reply:
Yeah, we can resolve this!